pkg/apis/planetscale/v2/vitessshard_methods.go

```diff
@@ -151,3 +153,47 @@ func (s *VitessShardSpec) CellInCluster(cellName string) bool {
 	_, inZoneMap := s.ZoneMap[cellName]
 	return inZoneMap
 }
+
+// NewVitessShardCondition returns a VitessShardCondition object based on the provided type and initial state.
+func NewVitessShardCondition(ty VitessShardConditionType, initState corev1.ConditionStatus) *VitessShardCondition {
+	now := v1.NewTime(time.Now())
+	return &VitessShardCondition{
+		Type:               ty,
+		Status:             initState,
+		LastTransitionTime: &now,
+		Reason:             "initState",
+		Message:            "The initial state for this VitessShardCondition.",
+	}
+}
+
+// ChangeStatus changes the status if the current status is not the same as the new status, and updates the
+// last transition time.
+func (c *VitessShardCondition) ChangeStatus(newStatus corev1.ConditionStatus, reason, message string) {
```

Comment:
SetStatus feels more natural to me, but I don't feel strongly.

Reply:
I had named it `ChangeStatus` because it bailed immediately if the statuses were the same. Now that we set the reason and message every time regardless, I think `SetStatus` is a better name. I'll change it.

Reply:
Updated: https://github.com/planetscale/vitess-operator/pull/31/commits/89eb420dbd304ab7b6a9b440d46fc4432a93127f.